packages/react-vapor/src/components/modal/tests/ConfirmationModalProvider.spec.tsx

```diff
@@ -1,155 +1,134 @@
-        expect(() => {
-            shallowMountUnsavedModalProvider().unmount();
-        }).not.toThrow();
-    });
-
-    describe('when should confirm changes', () => {
-        it('should open the ConfirmationModal to interrupt the leaving action', () => {
-            heavyConfirmationModalProvider = heavilyMountUnsavedModalProvider(true);
-
-            expect(heavyConfirmationModalProvider.childAt(1).prop('isOpen')).toBeFalsy();
-
-            const heavyModal: any = heavyConfirmationModalProvider.find(ModalCompositeConnected);
-            heavyModal.props().modalFooterChildren.props.onClick();
-            heavyConfirmationModalProvider.update();
-
-            expect(heavyConfirmationModalProvider.childAt(1).prop('isOpen')).toBeTruthy();
-        });
-
-        describe('PromptBefore function', () => {
-            it('should return false', () => {
-                confirmationModalProvider = shallowMountUnsavedModalProvider(true);
-                confirmationModalProvider.childAt(0).props().modalFooterChildren.props.onClick();
-
-                expect(regularClickActionSpy).not.toHaveBeenCalled();
-            });
+                    )}
+                </ConfirmationModalProvider>,
+                {initialState: {modals: [{id: '👾', isOpened: true}]}}
+            );
+            expect(screen.queryByText('Are you sure?')).not.toBeInTheDocument();
+            userEvent.click(screen.getByRole('button', {name: 'here'}));
+            expect(screen.getByText('Are you sure?')).toBeVisible();
         });
 
-        it('should use the callback function in the promptBefore function when confirming the exit', () => {
-            heavyConfirmationModalProvider = heavilyMountUnsavedModalProvider(true);
-
-            const modal: any = heavyConfirmationModalProvider.find(ModalCompositeConnected);
-            modal.props().modalFooterChildren.props.onClick();
-            heavyConfirmationModalProvider.update();
-            const exitWithoutSavingButton: any = heavyConfirmationModalProvider.childAt(1).find('Button').first();
-            exitWithoutSavingButton.prop('onClick')();
-
-            expect(promptBeforeClickActionSpy).toHaveBeenCalled();
-        });
-
-        it('should not use the callback function in the promptBefore function when cancelling the exit', () => {
-            heavyConfirmationModalProvider = heavilyMountUnsavedModalProvider(true);
-
-            const modal: any = heavyConfirmationModalProvider.find(ModalCompositeConnected);
-            modal.props().modalFooterChildren.props.onClick();
-            heavyConfirmationModalProvider.update();
-            const exitWithoutSavingButton: any = heavyConfirmationModalProvider.childAt(1).find('Button').last();
-            exitWithoutSavingButton.prop('onClick')();
-
-            expect(promptBeforeClickActionSpy).not.toHaveBeenCalled();
+        it('should open the ConfirmationModal to interrupt the leaving action if isDirty is false', () => {
```

Comment:
This test doesn't test any different functionality of the ConfirmationModalProvider compared to the test just above. We can safely remove it.